lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt

```diff
@@ -0,0 +1,87 @@
+package org.konigsoftware.kontext
+
+import com.google.protobuf.Message
+import io.grpc.Context
+import kotlin.reflect.KClass
+import kotlin.reflect.full.functions
+import kotlin.reflect.jvm.javaMethod
+
+class KonigKontext<KontextType>(
+    internal val defaultValue: KontextType,
+    internal val valueToBinary: (KontextType) -> ByteArray,
```

Comment:
these feel equally integral to the interface as `get()` and `setValue()` ? 

if so, maybe could define them alongside get/set? (probably need to convert this to an abstract class/interface?)

Reply:
..alternatively, I could see this class being split into two things/interfaces:

`KonigKontextSerializer`

`KonigRequestKontext`

(or similar)

Reply:
Addressed by adding the `KonigKontextKey` abstract class